### PR TITLE
es5 compatibility

### PR DESCRIPTION
**Pull request by gregplaysguitar**
Since this code isn't transpiled, it should use es5 syntax for browser compatibility

### CODE REVIEW

**Comment by samueljseay** [comment]
+1 for this, we use postcss-js in browser which depends on this package and would really like to avoid having to transpile this. 😄

**Comment by stevenvachon** [comment]
@samueljseay PostCSS is not ES5, so you already have to transpile.

**Comment by stevenvachon** [comment]
I'm still against releasing ES5 libraries. Transpiling a dependency from ES2015 to ES5 loses advantages and is no different from doing the reverse (with [lebab](https://npmjs.com/lebab) or [uglify-es](https://npmjs.com/uglify-es)).

**Comment by gregplaysguitar** [comment]
Ok, that's your prerogative; just be aware that we're now going to have to either convince other libs (i.e. postcss-js) not to depend on yours, or inline the code for those libraries, so this seems like a small change with a real benefit.

**Comment by ai** [comment]
@stevenvachon hi, we need ES5 support for PostCSS. Maybe you can reconsider it?

I think we should be focused on the user first, and only then on developers.

ES6 vs ES6 is not a big deal for 30 lines file. But it is a big deal for users, who can’t update their browsers. There are African users without any computer expect the old non-smartphone phone, which supports only Opera Mini. There are IE users, which can’t update it because they work from Internet cafe.

I agree that in every case it is a question of balance. And I understand when big complex applications start to serve only ES6 compatible JS. But 30 lines dependency is not this case. It is just 5 minutes work (which already done) for you and really hours of help to users without ability to update browsers.

**Comment by Nedomas** [comment]
+1. It is also breaking the standard `yarn build` if you use it in the `create-react-app`.

File: index.js [hunk 1]
```diff
@@ -1,12 +1,11 @@
...
+var pattern = /-(\w|$)/g;
 
-const callback = (dashChar, char) => char.toUpperCase();
+var callback = (dashChar, char) => char.toUpperCase();
 
 
 
-const camelCaseCSS = property =>
-{
+var camelCaseCSS = function (property) {
```
Comment (stevenvachon) [inline index.js]:
```js
function camelCaseCSS(property)
{
```

File: index.js [hunk 2]
```diff
@@ -1,12 +1,11 @@
 "use strict";
-const pattern = /-(\w|$)/g;
+var pattern = /-(\w|$)/g;
 
-const callback = (dashChar, char) => char.toUpperCase();
+var callback = (dashChar, char) => char.toUpperCase();
```
Comment (stevenvachon) [inline index.js]: This will need to be a regular function. With that, I'd prefer not a variable declaration.

**Review by stevenvachon — CHANGES_REQUESTED** [review]

**Comment by stevenvachon** [comment]
@ai doesn't PostCSS use Promises?

**Comment by ai** [comment]
@stevenvachon yes, but end-users can add polyfill for Promise.

**Comment by stevenvachon** [comment]
@ai end-users can also run a global transform on their codebase. Similar difference.

**Comment by stevenvachon** [comment]
How do you all feel about the use of ES2015's `String#startsWith`? It will need a polyfill.

**Comment by ai** [comment]
Global compilation is much harder, most of developers will afraid of do it.

I try to not use `startsWith`. Send PR if you will find it.

I have a responsibilities about support. This talk will not change them, because I already promised support. So it is not just my thoughts. I have a exact problem.

If you don't want to support big projects and care about compatibility, it is OK. It  doesn't make a bad person. We all just do your hobbies. Just say that this project is just for fun and we will replace dependency or inline code of it.

I don't need to change your mind, I just need to fix exact problem.

**Comment by stevenvachon** [comment]
Like all of my projects, I wrote them for myself and I share them in an ideal state. There's no point in duplicating code in the world, so I'll conform this library to its needs. I don't want to sacrifice code quality, though, and switching from `startsWith` to `indexOf` is affecting that. I think I'll just add a build step.

**Comment by ai** [comment]
Build step will not work with `startsWith` since it need a polyfill, not only compiling

**Comment by stevenvachon** [comment]
https://www.npmjs.com/package/babel-plugin-optimize-starts-with

**Comment by ai** [comment]
Wow, cool. Didn't know about it. Thanks for the link.

**Comment by stevenvachon** [comment]
I've just pushed the changes to master branch. Let me know if it solves the issue(s) and I'll release it.

**Comment by stevenvachon** [comment]
@ai, @gregplaysguitar, @Nedomas

**Comment by ai** [comment]
Sorry, I on the Def Con right now

**Comment by gregplaysguitar** [comment]
`index-es5.js` looks good to me. Just so I'm not misunderstanding, we would import like this? 

`const camelCaseCSS = require('camelcase-css/index-es5');`

**Comment by ai** [comment]
@gregplaysguitar nope, webpack will replace `index.js` to `index-es5` because of `browser` field in `package.json`

**Comment by ai** [comment]
LGTM, too :+1:

**Comment by stevenvachon** [comment]
Released as v2.0.1

Thanks!

**Pull request closed without merging**